lib/signs_ui/alerts/events.ex

```diff
@@ -0,0 +1,48 @@
+defmodule SignsUi.Alerts.Events do
+  alias SignsUi.Alerts.Alert
+  alias SignsUi.Alerts.State
+
+  @spec parse(String.t() | list() | map()) :: [State.row()] | State.row()
```

Comment:
I know they say the Elixir philosophy is "let it crash", but in my experience that's not actually a great solution. Here, what I imagine is if alerts-ui produces some garbage alert for some reason, that could get signs-ui stuck in a crash loop.

Since the data is coming from outside this application's boundary, I think we need to be a little more careful with it, and have this top level function, return `{:ok, State.row()} | {:ok, [State.row()]} | :error`, and kind of be a little more precise with the error handling throughout.

A lot of the `{:ok, _} = ...` chains you have, can be made fallible with the `with` expression:

```ex
with 
  {:ok, created_at_local, _} <- DateTime.from_iso8601(attributes["created_at"]),
  {:ok, created_at_utc} <- DateTime.shift_zone,
  routes <- parse_routes(attributes)
do
  {:ok, {....}}
else
 _ -> :error
end
```

Although, come to think of it as I type that up, doesn't `DateTime.from_iso8601` already return the DateTime in UTC?

Reply:
That's a good point, I didn't realize ISO8601 doesn't contain offset information.

Reply:
Well, I think it can, but `from_iso8601` will shift it for you:

```
iex> {:ok, datetime, 9000} = DateTime.from_iso8601("2015-01-23T23:50:07,123+02:30")
iex> datetime
~U[2015-01-23 21:20:07.123Z]
```